src/libs/OptionsListUtils.js

```diff
@@ -105,6 +125,7 @@ function createOption(personalDetailList, report, draftComments, {showChatPrevie
         icons: report ? report.icons : [personalDetail.avatar],
         tooltipText,
         participantsList: personalDetailList,
+        participantNames,
```

Comment:
Ok so, I think this should not be stored on the option itself. The options are used by the React components, but this information here is _only used when building the options_. It doesn't make a ton of sense to do in `createOption()` as we will never use this data again after determining whether to add the option or not.

My suggestion would be to remove this logic from here and put it in a new method called `getParticipantNames()` that takes a param of `personalDetailList` and returns the `participantNames`. If we have a search string then we can call this method and see if there is a match. If we don't have a search string then we don't really need to call `getParticipantNames()` anyway.

Reply:
Totally makes sense, thanks for the detailed explanation! Great to see how this could be cleaned up.

Reply:
If this is the case then I think I can still use a Set here, I think using an Object as a Set is kind of obscure.